### PR TITLE
Deleted account status, base free from WCPay settings and migrated few components from WCPay settings to overview page.

### PR DESCRIPTION
**Pull request by naman03malhotra**
Derived from: #1412 
Context: https://github.com/Automattic/woocommerce-payments/issues/1412#issuecomment-849894424

#### Changes proposed in this Pull Request
- Deleted account status and base fee from settings
- Migrated `ExpirationBar`, `ExpirationDescription`, and `LearnMoreLink` from WCPay settings (account-fee-settings) to the overview page.

#### Testing instructions

- switch to `migrate/account-fees-and-tests`.
- Goto WCPay settings.
- Account status and Base fee should not render.
- Goto Overview page.
- Pass dummy data (Ref. https://github.com/Automattic/woocommerce-payments/pull/1966/files#discussion_r642142580)
- It should show the expiration bar and description for promo.

#### After migration:
image

image

image

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

### CODE REVIEW

File: client/components/account-status/account-fees/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	const feeDescription = formatAccountFeesDescription( accountFees );
+	const currentFee = getCurrentFee( accountFees );
+
+	return (
+		<>
+			{ currencyName ? `${ currencyName } ` : null }
+			{ currencyCode ? `(${ currencyCode }) ` : null }
+			{ feeDescription }
+			<ExpirationBar feeData={ currentFee } />
+			<ExpirationDescription feeData={ currentFee } />
```
Comment (naman03malhotra) [inline client/components/account-status/account-fees/index.js:49]: For testing:
```suggestion
			<ExpirationBar feeData={ { volume_allowance: 100, current_volume: 60 } } />
			<ExpirationDescription feeData={ { volume_allowance: 100, current_volume: 60 } } />
```

File: includes/class-wc-payment-gateway-wcpay.php [hunk 2]
```diff
@@ -1162,54 +1156,6 @@ public function generate_checkbox_html( $key, $data ) {
...
-		return ob_get_clean();
-	}
-
-	/**
-	 * Generates markup for the fees information section.
-	 *
-	 * @return string Markup or empty if the account is not connected.
-	 */
-	public function generate_account_fees_html() {
-		if ( ! $this->is_connected() || empty( $this->account->get_fees() ) ) {
```
Comment (RadoslavGeorgiev) [inline includes/class-wc-payment-gateway-wcpay.php:1195]: Any idea why the fees object might ever be empty? Do we need to handle it in JS as well?

File: client/components/account-status/account-fees/index.js (hunk 1 above)
Comment (RadoslavGeorgiev) [reply client/components/account-status/account-fees/index.js:53]: Works well with real data too 🥳 

image

**Review by RadoslavGeorgiev — APPROVED** [review]
Thank you @naman03malhotra, the changes look good and work well!

The origin of this is not in this PR, but I think it would be a good place to fix it: There's missing space and instead of `Base Fee` the status item about fees says `BaseFee`.

File: includes/class-wc-payment-gateway-wcpay.php (hunk 2 above)
Comment (naman03malhotra) [reply includes/class-wc-payment-gateway-wcpay.php:1195]: Not sure about this, ideally it never should be.
Any idea @Automattic/gamma ?

File: client/components/account-status/account-fees/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	const currencyCode = currency?.getCurrencyConfig()?.code;
+	const feeDescription = formatAccountFeesDescription( accountFees );
+	const currentFee = getCurrentFee( accountFees );
+
+	return (
+		<>
+			{ currencyName ? `${ currencyName } ` : null }
+			{ currencyCode ? `(${ currencyCode }) ` : null }
+			{ feeDescription }
+			<ExpirationBar feeData={ currentFee } />
```
Comment (RadoslavGeorgiev) [inline client/components/account-status/account-fees/index.js:52]: Both `<ExpirationBar />` and `<ExpirationDescription />` below default to `USD` for their currency. Let's somehow include the currency from the base fee for them.

**Review by RadoslavGeorgiev — CHANGES_REQUESTED** [review]
I mentioned this in Slack, but I'm adding it here for posterity.

And the bigger question (not for this PR): How do promotions and MDCs work together?

File: client/components/account-status/account-fees/index.js (hunk 3 above)
Comment (naman03malhotra) [reply client/components/account-status/account-fees/index.js]: Thanks for catching this! fixed it in https://github.com/Automattic/woocommerce-payments/pull/1966/commits/4f4381ab79af2063b2748ee878ee40a7e2b96630.

**Comment by dmvrtx** [comment]
@naman03malhotra, I think you've managed to solve #1376 with 8952edb24abfa5c3be2dc292981b2b9b2614be2d 😉

**Comment by naman03malhotra** [comment]
>@naman03malhotra, I think you've managed to solve #1376 with 8952edb 😉

Hahaha, nice...thanks for noticing @dmvrtx 🙂 .

**Comment by naman03malhotra** [comment]
> There's missing space and instead of Base Fee the status item about fees says BaseFee.
Just noticed this, fixed it in https://github.com/Automattic/woocommerce-payments/pull/1966/commits/9d655d7d9422b4c1f83921de177c1d82eefcf1b2

**Review by RadoslavGeorgiev — APPROVED** [review]
LGTM again! Thank you for fixing the currencies!

image

File: includes/class-wc-payment-gateway-wcpay.php (hunk 2 above)
Comment (kalessil) [reply includes/class-wc-payment-gateway-wcpay.php:1195]: Currently, the fees will always be provided.

The original code was both server and client changes, so likely it's just an assertive construct for the case the server part is getting reverted.

**Comment by kalessil** [comment]
LGTM !

**Pull request merged**